data_steward/deid/aou.py

```diff
@@ -298,24 +293,36 @@ def initialize(self, **args):
             self.map_questionnaire_response_ids(million)
             create_person_id_src_hpo_map(self.idataset, self.credentials)
 
-        if person_table.shape[0] > 0:
-            person_table = person_table[person_table.age < age_limit]
-            map_sql = 'SELECT * FROM ' + map_tablename
-            job_config = {
-                'query': {
-                    'defaultDataset': {
-                        'datasetId': self.idataset
-                    }
-                }
-            }
-            map_table = self.get_dataframe(sql=map_sql, query_config=job_config)
-            if not map_table.shape[0] > 0:
-                LOGGER.error(
-                    "Unable to initialize Deid. _deid_map has no participants to load."
-                )
+        # ensure mapping table only contains participants within age limits
+        sql = (f"SELECT DISTINCT p.person_id, "
+               f"EXTRACT(YEAR FROM CURRENT_DATE()) - year_of_birth AS age "
```

Comment:
How strict do we need to be for the age cutoff? I'm wondering if `birth_datetime` is a more accurate cutoff. If it is null, we could resort to using `year_of_birth` (possibly along with `day_of_birth` and `month_of_birth` unless they are null as well). What are your thoughts on this?

Reply:
I think I want to leave this one alone.  This is how Steve wrote Deid before it got handed off to us.  I like this approach because `year_of_birth` is a required field while `birth_datetime` is nullable.